src/main/java/com/github/jikoo/regionerator/hooks/BentoBoxHook.java

```diff
@@ -0,0 +1,39 @@
+package com.github.jikoo.regionerator.hooks;
+
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.jetbrains.annotations.NotNull;
+
+import world.bentobox.bentobox.BentoBox;
+import world.bentobox.bentobox.managers.IslandsManager;
+
+
+/**
+ * PluginHook for <a href="https://github.com/BentoBoxWorld/BentoBox">BentoBox</a> and its GameModes.
+ */
+public class BentoBoxHook extends PluginHook
+{
+	public BentoBoxHook()
+	{
+		super("BentoBox");
+	}
+
+	@Override
+	public boolean isChunkProtected(@NotNull World chunkWorld, int chunkX, int chunkZ)
+	{
+		if (!BentoBox.getInstance().getIWM().inWorld(chunkWorld))
+		{
+			return false;
+		}
+
+		int locX = chunkX << 4;
+		int locZ = chunkZ << 4;
+
+		IslandsManager manager = BentoBox.getInstance().getIslandsManager();
+
+		return manager.getIslandAt(new Location(chunkWorld, locX, 0, locZ)).isPresent() ||
+			manager.getIslandAt(new Location(chunkWorld, locX + 15, 0, locZ)).isPresent() ||
+			manager.getIslandAt(new Location(chunkWorld, locX + 15, 0, locZ + 15)).isPresent() ||
+			manager.getIslandAt(new Location(chunkWorld, locX, 0, locZ + 15)).isPresent();
```

Comment:
Technically yes, practically no.

Islands with distance smaller then 16 blocks between them are possible by config, but impractical in gameplay.

Reply:
Is there a way to check this config without parsing it ourselves? Even if it is impractical, opening the door to wrongly deleting someone's stuff isn't something I'm comfortable with.
Throwing an IllegalStateException in the constructor if the minimum size is < 16 would cause Regionerator to log the issue and pause deletion until the user manually intervened and fixed their setup.